File: mirea/contests/2021-09-26 MIREA Welcome Contest 2021/NoPoint C++/Задача E O(n^2).cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;
#define int ll

int32_t main() {
  int tests;
  cin >> tests;
  for(int test = 0; test < tests; ++test) {
    int n;
    cin >> n;
    vector<int> arr(n);
    for(int i = 0; i < n; ++i) {
      cin >> arr[i];
    }

    int ans = 3e18;
    for(int split = 1; split < n; ++split) {
      int one = arr[split - 1];
      int other = arr[split];
      int p1 = split - 2;
      int p2 = split + 1;
      ans = min(ans, abs(one - other));
      while(true) {
        if (one < other) {
          if (p1 >= 0) {
            one += arr[p1--];
          } else {
            break;
          }
        } else {
          if (p2 < n) {
            other += arr[p2++];
          } else {
            break;
          }
        }
        ans = std::min(ans, abs(one - other));
      }
    }
    cout << ans << endl;
  }
}

File: mirea/contests/2021-09-26 MIREA Welcome Contest 2021/NoPoint Python/Задача E O(n^2).py
# Сдавать под pypy3
def main():
  tests = int(input())

  for test in range(tests):
    n = int(input())
    arr = [int(num) for num in input().split()]
    ans = 10 ** 18
    for split in range(1, n):
      one = arr[split - 1]
      other = arr[split]
      p1 = split - 2
      p2 = split + 1
      ans = min(ans, abs(one - other))
      while True:
        if one < other:
          if p1 >= 0:
            one += arr[p1]
            p1 -= 1
          else:
            break
        else:
          if p2 < n:
            other += arr[p2]
            p2 += 1
          else:
            break
        ans = min(ans, abs(one - other))
    print(ans)


main()

